mmpose/datasets/pipelines/gesture_transform.py

```diff
@@ -0,0 +1,400 @@
+            results['width'][i], results['height'][i] = width, height
+        return results
+
+
+@PIPELINES.register_module()
+class MultiFrameBBoxMerge:
+    """Compute the union of bboxes in selected frames.
+
+    Required keys: 'bbox'.
+
+    Modified keys: 'bbox'.
+    """
+
+    def __init__(self):
+        pass
+
+    def __call__(self, results):
+        if 'bbox' not in results:
+            return results
+
+        bboxes = list(filter(lambda x: len(x), results['bbox']))
+        if len(bboxes) == 0:
+            bbox_xyxy = np.array(
+                (0, 0, results['width'][0] - 1, results['height'][0] - 1))
+        else:
+            bboxes_xyxy = np.stack([b[0]['bbox'] for b in bboxes])
+            bbox_xyxy = np.array((
+                bboxes_xyxy[:, 0].min(),
+                bboxes_xyxy[:, 1].min(),
+                bboxes_xyxy[:, 2].max(),
+                bboxes_xyxy[:, 3].max(),
+            ))
+        results['bbox'] = bbox_xyxy
+        return results
+
+
+@PIPELINES.register_module()
+class ResizedCropByBBox:
+    """Spatial crop for spatially aligned videos by bounding box.
+
+    Required keys: 'video', 'modality', 'width', 'height', 'bbox'.
+
+    Modified keys: 'video', 'width', 'height'.
+    """
+
+    def __init__(self, size, scale=(1, 1), ratio=(1, 1), shift=0):
+        self.size = size if isinstance(size, (tuple, list)) else (size, size)
+        self.scale = scale
+        self.ratio = ratio
+        self.shift = shift
+
+    def __call__(self, results):
+        bbox_xywh = bbox_xyxy2xywh(results['bbox'][None, :])[0]
+        length = bbox_xywh[2:].max()
+        length = length * np.random.uniform(*self.scale)
+        x = bbox_xywh[0] + np.random.uniform(-self.shift, self.shift) * length
+        y = bbox_xywh[1] + np.random.uniform(-self.shift, self.shift) * length
+        w, h = length, length * np.random.uniform(*self.ratio)
+
+        bbox_xyxy = bbox_xywh2xyxy(np.array([[x, y, w, h]]))[0]
+        bbox_xyxy = bbox_xyxy.clip(min=0)
+        bbox_xyxy[2] = min(bbox_xyxy[2], results['width'][0])
+        bbox_xyxy[3] = min(bbox_xyxy[3], results['height'][0])
+        bbox_xyxy = bbox_xyxy.astype(np.int32)
+
+        for i in range(len(results['video'])):
+            video = results['video'][i].transpose(1, 2, 3, 0)
+            num_frames = video.shape[-1]
+            video = video.reshape(video.shape[0], video.shape[1], -1)
+            video = mmcv.imcrop(video, bbox_xyxy)
+            video = mmcv.imresize(video, self.size)
+
+            results['video'][i] = video.reshape(video.shape[0], video.shape[1],
+                                                -1, num_frames)
+            results['video'][i] = results['video'][i].transpose(3, 0, 1, 2)
+            results['width'][i], results['height'][i] = video.shape[
+                1], video.shape[0]
+
+        return results
+
+
+@PIPELINES.register_module()
+class NVGestureRandomFlip:
```

Comment:
It seems that this pipeline is specific to NVGesture dataset? Can we make it more general?

Reply:
Ok. Horizontal flip sometimes changes the category of the input video. This pipeline needs to know the label pairs that represent lateral symmetric gestures. How about making such label pairs a parameter for this pipeline, or adding `flipped` and `label_after_flip` items in `results`?

Reply:
This is similar to https://github.com/open-mmlab/mmpose/blob/1e8f1e53cf8218fa575a1b2bdccdf52f9ac8adbb/mmpose/datasets/pipelines/top_down_transform.py#L129